test/url-to-options.js

```diff
@@ -0,0 +1,56 @@
+import url from 'url';
+import test from 'ava';
+import urlToOptions from '../source/url-to-options';
+
+const githubURL = 'https://user:password@github.com:443/';
```

Comment:
It should be more complicated. Like:

```js
const githubURL = 'https://user:password@github.com:443/say?hello=world#awesome';
```

Also I would rename `githubURL` to `exampleURL`.